tests/integration/solidity_contracts/Solmate/test_erc20.py

```diff
@@ -80,6 +72,20 @@ async def test_should_transfer(self, erc_20, owner, other):
             assert await erc_20.balanceOf(owner.address) == 0
             assert await erc_20.balanceOf(other.address) == TEST_SUPPLY
 
+        async def test_transfer_should_fail_when_insufficient_balance(
+            self, erc_20, owner, other
+        ):
+            await erc_20.mint(
+                owner.address, TEST_AMOUNT, caller_address=owner.starknet_address
+            )
+            with kakarot_error():
```

Comment:
are we asserting on the error message itself? 
as was done before:  `assert message == "Kakarot: Reverted with reason: 12884901888"`

Reply:
the numbers returned before didn't necessarily have sense. It's possible that the code revert with no reason actually, and it's often what is used in Solmate's contrat (see the `require` are only used for `permit`).


They use the fact that underflow will revert to avoid writing `require` themselves and in this case, the `revert_reason` is `[]` (size = 0).